nilearn/connectome/connectivity_matrices.py

```diff
@@ -231,6 +231,42 @@ def sym_to_vec(symmetric, discard_diagonal=False):
     return symmetric[..., tril_mask] / scaling[tril_mask]
 
 
+def vec_to_sym(vec):
+    """Return the symmetric array given its flattened lower triangular part.
+    Acts on the last dimension of the array if not 1-dimensional.
+    Parameters
+    ----------
+    vec : numpy.ndarray, shape (..., p * (p + 1) /2)
+        The input array.
+
+    Returns
+    -------
+    sym : numpy.ndarray, shape (..., p, p)
+        The output symmetric array.
+    """
+    n = vec.shape[-1]
+    # solve p * (p + 1) / 2 = n subj. to p > 0
+    # p ** 2 + p - 2n = 0 & p > 0
+    # p = - 1 / 2 + sqrt( 1 + 8 * n) / 2
+    p = (sqrt(8 * n + 1) - 1.) / 2
+    if p > floor(p):
+        raise ValueError(
+            "Vector of unstuitable shape {0} can not be transformed to "
+            "a symmetric matrix.".format(vec.shape))
+
+    p = int(p)
+    mask = np.tril(np.ones((p, p))).astype(np.bool)
+    sym = np.zeros(vec.shape[:-1] + (p, p))
+    sym[..., mask] = vec.copy()  # avoid side effects
+    sym.swapaxes(-1, -2)[..., mask] = vec
```

Comment:
> OK. Does it make sense to make vec_to_sym private? I don't think it will be used apart from here

I think that people (methods developers) might use it to build their own
algorithms. As nilearn is also for methods developers, I think that it
makes sense to expose it publicaly.